src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py

```diff
@@ -663,5 +663,48 @@ def on_goto_request(self, py_db, request):
         response = pydevd_base_schema.build_response(request, kwargs={'body': {}})
         return NetCommand(CMD_RETURN, 0, response, is_json=True)
 
+    def _can_set_dont_trace_pattern(self, py_db, start_patterns, end_patterns):
+        if py_db.is_cache_file_type_empty():
+            return True
+
+        if py_db.dont_trace_external_files.__name__ == 'dont_trace_files_property_request':
+            return py_db.dont_trace_external_files.start_patterns == start_patterns and \
+                py_db.dont_trace_external_files.end_patterns == end_patterns
+
+        return True
+
+    def on_setdebuggerproperty_request(self, py_db, request):
+        args = request.arguments.kwargs
+        if 'dontTraceStartPatterns' in args and 'dontTraceEndPatterns' in args:
```

Comment:
This feels like it would be more straightforward with a regex instead (which could easily cover both cases, and then some). Or is there a perf issue with them on a hot path here?

Reply:
I don't have a preference here. I can change it either way. For now I will be making an update to this where we will be disabling this stuff. temporarily until we address #1355